barretenberg/cpp/src/barretenberg/dsl/acir_format/aes128_constraint.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+#include "barretenberg/dsl/types.hpp"
+#include "barretenberg/serialize/msgpack.hpp"
+#include <cstdint>
+#include <vector>
+
+namespace acir_format {
+
+struct AES128Input {
```

Comment:
can the num_bits for each of the input vector values be fixed, given it is padded on the noir side?

Reply:
It's padded to a multiple of 16 bytes but we don't really know the total length...we do know each value is 8bit, is that what you mean?

Reply:
Ignoring the first comment, as i realised the num_bits field is required 

edit is that the AES128Input is pretty much the same as the FunctionInput type, was wondering if it could be used instead
<img width="1146" alt="image" src="https://github.com/AztecProtocol/aztec-packages/assets/47148561/86eacfa8-7397-448f-9422-babb89e02ad4">

Reply:
I tried using it directly, but a bunch of the serde autogenerated code started complaining about unused parameters...I think I'm gonna leave it like that since it can be refactored along all the other constraint definitions that seem to use custom structs too.